src/R/Editor/Impl/Completion/RFunctionCompletion.cs

```diff
@@ -36,9 +35,11 @@ public class RFunctionCompletion : RCompletion {
         }
 
         private void TryFetchDescription() {
-            Task.Run(async () => {
-                SetDescription(await _functionIndex.GetFunctionInfoAsync(this.DisplayText));
-            }).Wait(500);
+            _functionIndex.GetFunctionInfoAsync(this.DisplayText).ContinueWith(t => {
+                if (t.IsCompleted) {
```

Comment:
`Completed` means that it is either `IsFaulted` or `IsCanceled` or `Status == RunToCompletion`